src/plugins/kibana/public/dashboard/components/panel/panel.html

```diff
@@ -1,14 +1,31 @@
 <div class="panel panel-default" ng-switch on="panel.type" ng-if="savedObj || error">
   <div class="panel-heading">
-    <span class="panel-title">
+    <div class="panel-title">
       <i
         class="fa"
         ng-class="savedObj.vis.type.icon"
         aria-label="{{savedObj.vis.type.title}} Icon"
         title="{{savedObj.vis.type.title}}">
       </i>
-      {{savedObj.title}}
-    </span>
+      <a class="action panel-title-edit" ng-click="setTitle()">
+	      <span ng-if="panel.customTitle">{{panel.customTitle}}</span>
+	      <span ng-hide="panel.customTitle">{{savedObj.title}}</span>
+      </a>
+      <input size="10" ng-if="showFormTitle" text="title" ng-model="panel.customTitle" ng-show="chrome.getVisible() && editUrl">
```

Comment:
- This needs the proper bootstrap form classes added to it. See examples in the query bar.
- Requires editUrl in order to be shown? 
- Use uiState w/ get/set instead of panel
- Just call it title instead of customTitle
- Don't set size=10, use css to control the input width. See bootstrap way of doing this.
- The ng-click shouldn't fire if !chrome.getVisible()